api/routes/apds/activities/post.js

```diff
@@ -0,0 +1,77 @@
+const logger = require('../../../logger')('apd activites route put');
+const {
+  apdActivity: defaultActivityModel,
+  apd: defaultApdModel
+} = require('../../../db').models;
+const { userCanEditAPD: defaultUserCanEditAPD } = require('../utils');
+const loggedIn = require('../../../auth/middleware').loggedIn;
+
+module.exports = (
+  app,
+  ActivityModel = defaultActivityModel,
+  ApdModel = defaultApdModel,
+  userCanEditAPD = defaultUserCanEditAPD
+) => {
+  const check404 = async req => {
+    const apdID = +req.params.apdID;
+
+    if (!await userCanEditAPD(req.user.id, apdID)) {
+      logger.verbose(req, 'user (state) not associated with this apd');
+      return false;
+    }
+
+    const apd = await ApdModel.where({ id: apdID }).fetch({
+      withRelated: ['activities']
+    });
+    if (!apd) {
+      logger.verbose(req, `no such apd [${apdID}]`);
+      return false;
+    }
+
+    return apd;
+  };
+
+  logger.silly('setting up POST /apds/:apdID/activities route');
+  app.post('/apds/:apdID/activities', loggedIn, async (req, res) => {
+    logger.silly(req, 'handling POST /apds/:apdID/activities route');
+    logger.silly(
+      req,
+      `attempting to add an activity on apd [${req.params.apdID}]`
+    );
+
+    try {
+      const apd = await check404(req, res);
```

Comment:
i like this `check404` function, but it's kinda weird (to me at least) that it returns either false or an apd instance (the latter of which surprises me given the function name), but i'm not sure of a better way without a duplicative lookup; i think maybe just a name tweak would help, is `check404AndGetApd` terrible?

Reply:
I'm glad you caught that, I meant to go back and try to figure out a better name and/or return type.  The reason for `false` is just to be explicit that it's not returning an APD, but returning `null` would be fine too.  Then it could maybe just be `getApd()`?

Reply:
`getApd` is great